--- django/core/management/commands/dumpdata.py.bak	2012-08-17 21:05:29.195189389 -0500
+++ django/core/management/commands/dumpdata.py	2012-08-20 10:28:04.808474184 -0500
@@ -147,11 +147,11 @@
             for field in model._meta.fields:
                 if hasattr(field.rel, 'to'):
                     rel_model = field.rel.to
-                    if hasattr(rel_model, 'natural_key'):
+                    if hasattr(rel_model, 'natural_key') and rel_model != model:
                         deps.append(rel_model)
             for field in model._meta.many_to_many:
                 rel_model = field.rel.to
-                if hasattr(rel_model, 'natural_key'):
+                if hasattr(rel_model, 'natural_key') and rel_model != model:
                     deps.append(rel_model)
             model_dependencies.append((model, deps))
 
